moshi/src/main/java/com/squareup/moshi/JsonUtf8Reader.java

```diff
@@ -241,10 +241,6 @@ public Token peek() throws IOException {
   }
 
   private int doPeek() throws IOException {
-    if (valueSource != null) {
-      valueSource.discard();
-      valueSource = null;
-    }
```

Comment:
I didn’t like how we were evaluating the state of valueSource on like, EVERY read.